src/overlays/actors/ovl_En_Ma_Yts/z_en_ma_yts.c

```diff
@@ -21,48 +74,521 @@ const ActorInit En_Ma_Yts_InitVars = {
+    }
+}
+
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+    { 18, 46, 0, { 0, 0, 0 } },
+};
+
+static CollisionCheckInfoInit2 D_80B8E19C = {
+    0, 0, 0, 0, 0xFF,
+};
+
+struct struct_80B8E1A8 {
+    /* 0x00 */ AnimationHeader* unk_00;
+    /* 0x04 */ s32 unk_04;
+    /* 0x08 */ u8 unk_08;
+    /* 0x0C */ f32 unk_0C;
+};
+
+static struct struct_80B8E1A8 D_80B8E1A8[] = {
+    { &D_06009E58, 0x3F800000, 0x00, 0.0f }, { &D_06009E58, 0x3F800000, 0x00, -6.0f },
+    { &D_06018948, 0x3F800000, 0x02, 0.0f }, { &D_06018948, 0x3F800000, 0x02, -6.0f },
+    { &D_0601B76C, 0x3F800000, 0x00, 0.0f }, { &D_0601B76C, 0x3F800000, 0x00, -6.0f },
+    { &D_06007328, 0x3F800000, 0x00, 0.0f }, { &D_06007328, 0x3F800000, 0x00, -6.0f },
+    { &D_06014088, 0x3F800000, 0x00, 0.0f }, { &D_06014088, 0x3F800000, 0x00, -6.0f },
+    { &D_06002A8C, 0x3F800000, 0x00, 0.0f }, { &D_06002A8C, 0x3F800000, 0x00, -6.0f },
+    { &D_06015B7C, 0x3F800000, 0x00, 0.0f }, { &D_06015B7C, 0x3F800000, 0x00, -6.0f },
+    { &D_06007D98, 0x3F800000, 0x00, 0.0f }, { &D_06007D98, 0x3F800000, 0x00, -6.0f },
+    { &D_0600852C, 0x3F800000, 0x00, 0.0f }, { &D_0600852C, 0x3F800000, 0x00, -6.0f },
+    { &D_06008F6C, 0x3F800000, 0x00, 0.0f }, { &D_06008F6C, 0x3F800000, 0x00, -6.0f },
+    { &D_060180DC, 0x3F800000, 0x02, 0.0f }, { &D_060180DC, 0x3F800000, 0x02, -6.0f },
+};
+
+static void* sMouthTextures[] = {
+    D_060127C8,
+    D_06012BC8,
+    D_06012FC8,
+    D_060133C8,
+};
+
+static void* sEyeTextures[] = {
+    D_0600FFC8, D_060107C8, D_06010FC8, D_060117C8, D_06011FC8,
+};
+
+static u16 D_80B8E32C = 99;
+
+void EnMaYts_ChangeAnim(EnMaYts* this, s32 index) {
+    SkelAnime_ChangeAnim(&this->skelAnime, D_80B8E1A8[index].unk_00, 1.0f, 0.0f,
+                         SkelAnime_GetFrameCount(&D_80B8E1A8[index].unk_00->common), D_80B8E1A8[index].unk_08,
+                         D_80B8E1A8[index].unk_0C);
+}
+
+void func_80B8D12C(EnMaYts* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    s16 flag = this->unk_32C == 2 ? true : false;
+
+    if ((this->unk_32C == 0) || (this->actor.parent == NULL)) {
+        this->unk_1D8.unk_18 = player->base.world.pos;
+        this->unk_1D8.unk_18.y -= -10.0f;
+    } else {
+        Math_Vec3f_StepTo(&this->unk_1D8.unk_18, &this->actor.parent->world.pos, 8.0f);
+        this->unk_1D8.unk_18.y -= -10.0f;
+    }
+
+    func_800BD888(&this->actor, &this->unk_1D8, 0, flag);
+}
+
+void EnMaYts_InitAnimation(EnMaYts* this, GlobalContext* globalCtx) {
+    switch (this->type) {
+        case EN_NA_YTS_TYPE_BARN:
+            this->actor.targetMode = 0;
+            EnMaYts_ChangeAnim(this, 0);
```

Comment:
This function sets an animation according to the parameter passed (which is an index of `D_80B8E1A8`).
I considered the idea of making macros/enums for the parameter of this function, but I don't know what we think about this. Suggestions?

Reply:
I think an enum would be fine either now or during object decomp later on, but they should represent the animationSeg accurately for where the animation data is pointing to. Just my thoughts